pages/profile/[username].tsx

```diff
@@ -0,0 +1,138 @@
+import { BaseContainer } from "../../component/Page/BaseContainer";
+import { Helmet } from "react-helmet";
+import React, { useEffect, useState } from "react";
+import LayoutContainer from "../../component/LayoutContainer";
+import Image from "next/image";
+import { Header } from "../../component/Header";
+import Toggle from 'react-toggle'
+import { useGetProfile } from "../../services/profile/profileHooks";
+import { useRouter } from "next/router";
+import { ITokenEnum } from "../../shared/enum";
+import { GetServerSidePropsContext } from "next";
+import { RotatingTriangles } from 'react-loader-spinner'
+import { toast } from "react-toastify";
+
+interface PageProps {
+    username?: string
+}
+
+export default function Profile(props: PageProps) {
+    const { data, isLoading } = useGetProfile(props.username);
+    const [message, setMessage] = useState('')
+    const router = useRouter();
+    useEffect(() => {
+        if (data) {
+            localStorage.setItem(ITokenEnum.targetUser, data.user_id);
+        }
+    }, [data])
+
+    return (
+        <BaseContainer className="bg-black">
+            <Helmet>
+                <title>{`${data?.profile_pic_path || ''}'s profile`}</title>
+            </Helmet>
+            <LayoutContainer backgroundImage="/image/background-profile.svg">
+                <Header />
+                <div className="flex flex-1-0-0 w-full justify-center items-center">
+                    {/* Card Header */}
+                    {isLoading ? <RotatingTriangles
+                        visible={true}
+                        height="80"
+                        width="80"
+                        ariaLabel="rotating-triangels-loading"
+                        wrapperStyle={{}}
+                        wrapperClass="rotating-triangels-wrapper"
+                    /> : <div className="flex w-full bg-white p-4 rounded-2xl flex-col gap-4 h-max">
```

Comment:
Can we make this to a loading wrapper?
e.g.
```
<LoaderWrapper isLoading={isLoading}>
<div className="flex ....>
....
</div>
</LoaderWrapper>
```

So the conditional loading is inside the loader wrapper.